language/extensions/async/move-async-vm/src/natives.rs

```diff
@@ -75,7 +75,7 @@ fn native_send(
     let ext = context.extensions_mut().get_mut::<AsyncExtension>();
     let mut bcs_args = vec![];
     while args.len() > 2 {
-        bcs_args.push(pop_arg!(args, Vector).to_vec_u8()?);
+        bcs_args.push(pop_arg!(args, Vec<u8>));
```

Comment:
Ah! That actually works, great. Problem with that kind of magic is that the APIs aren't clearly documented and buried somewhere in all those trait impls.

Reply:
agreed its a bit hard to discover, not sure if @vgao1996 has any thoughts on that (to be clear the VMValueCast stuff has been around a while, not new to this PR)